### ISSUE TITLE
About the testing time comsumption

### ISSUE THREAD

**Issue by RuijieJ** [ayumiymk/aster.pytorch#49]
According to the original paper, the recognition time for each image is 20ms. I modified demo.py, let it recognize all the images on the ICDAR2003 test set, and computed the average recognition time.

The following is my modification:

![image](https://user-images.githubusercontent.com/31977268/83383349-80c7c400-a417-11ea-8686-76d2ab67c5af.png)

and the output is 

```bash
average time = 168.360ms
```

which is larger than 20ms. I did the recognition on a single NVIDIA Tesla V100 GPU. I don't think using PyTorch will cause so much extra time consumption (I even ignore the time of reading image and preprocessing). So I wonder what is the problem.

Thanks very much.

**Comment by ayumiymk** [comment]
I guess there should be two reasons,

1. I am not very sure that the 20ms in our paper is whether tested with beam width 1.
2. The max_len in TF version is 30, see [here](https://github.com/bgshih/aster/blob/8d2be773fc17439ea76c0dbd229bc5284e61ce14/experiments/demo/config/trainval.prototxt#L97)? But in the pytorch version, 100 is used.

If you want to confirm the first point, you can run our TF code on a single NVIDIA TITAN Xp GPU with different beam widths.

**Comment by RuijieJ** [comment]
> I guess there should be two reasons,
> 
> 1. I am not very sure that the 20ms in our paper is whether tested with beam width 1.
> 2. The max_len in TF version is 30, see [here](https://github.com/bgshih/aster/blob/8d2be773fc17439ea76c0dbd229bc5284e61ce14/experiments/demo/config/trainval.prototxt#L97)? But in the pytorch version, 100 is used.
> 
> If you want to confirm the first point, you can run our TF code on a single NVIDIA TITAN Xp GPU with different beam widths.

Thanks. I found the slow test speed is caused by the first test sample. It took a long time to recognize the first sample, and the following samples will become faster. I ignore the test time of the first sample, and the average test time is 48ms (with beam width 1 and max_len 25). Although it is still larger than 20ms, it is acceptable.

**Comment by mengxiaolu** [comment]
> I guess there should be two reasons,
> 
> 1. I am not very sure that the 20ms in our paper is whether tested with beam width 1.
> 2. The max_len in TF version is 30, see [here](https://github.com/bgshih/aster/blob/8d2be773fc17439ea76c0dbd229bc5284e61ce14/experiments/demo/config/trainval.prototxt#L97)? But in the pytorch version, 100 is used.
> 
> If you want to confirm the first point, you can run our TF code on a single NVIDIA TITAN Xp GPU with different beam widths.



> > 我想应该有两个原因，
> > 
> > 1. 我不太确定论文中的20毫秒是否经过1毫米的光束宽度测试。
> > 2. TF版本中的max_len是30，请看[这里](https://github.com/bgshih/aster/blob/8d2be773fc17439ea76c0dbd229bc5284e61ce14/experiments/demo/config/trainval.prototxt#L97)？但是在pytorch版本中，使用了100。
> > 
> > 如果要确认第一点，则可以在具有不同光束宽度的单个NVIDIA TITAN Xp GPU上运行我们的TF代码。
> 
> 谢谢。我发现测试速度慢是由第一个测试样本引起的。花了很长时间才能识别出第一个样本，随后的样本会变得更快。我忽略了第一个样本的测试时间，平均测试时间为48ms（光束宽度为1，max_len为25）。尽管它仍然大于20ms，但是可以接受。

Hello, can the training and test code run through git bash, but can't run on the windows software platform?

Thank you very much.

**Comment by ayumiymk** [comment]
> > I guess there should be two reasons,
> > 
> > 1. I am not very sure that the 20ms in our paper is whether tested with beam width 1.
> > 2. The max_len in TF version is 30, see [here](https://github.com/bgshih/aster/blob/8d2be773fc17439ea76c0dbd229bc5284e61ce14/experiments/demo/config/trainval.prototxt#L97)? But in the pytorch version, 100 is used.
> > 
> > If you want to confirm the first point, you can run our TF code on a single NVIDIA TITAN Xp GPU with different beam widths.
> 
> > > 我想应该有两个原因，
> > > 
> > > 1. 我不太确定论文中的20毫秒是否经过1毫米的光束宽度测试。
> > > 2. TF版本中的max_len是30，请看[这里](https://github.com/bgshih/aster/blob/8d2be773fc17439ea76c0dbd229bc5284e61ce14/experiments/demo/config/trainval.prototxt#L97)？但是在pytorch版本中，使用了100。
> > > 
> > > 如果要确认第一点，则可以在具有不同光束宽度的单个NVIDIA TITAN Xp GPU上运行我们的TF代码。
> > 
> > 
> > 谢谢。我发现测试速度慢是由第一个测试样本引起的。花了很长时间才能识别出第一个样本，随后的样本会变得更快。我忽略了第一个样本的测试时间，平均测试时间为48ms（光束宽度为1，max_len为25）。尽管它仍然大于20ms，但是可以接受。
> 
> Hello, can the training and test code run through git bash, but can't run on the windows software platform?
> 
> Thank you very much.

Sorry, it only runs on Linux.

**Comment by lrfighting** [comment]
请问您知道如何在测试时候输出矫正图像吗？
